src/CodeGen_Hexagon.cpp

```diff
@@ -1349,6 +1349,8 @@ llvm::Function *CodeGen_Hexagon::define_hvx_intrinsic(llvm::Function *intrin,
                         internal_error
                             << "unhandled broadcast_scalar_word in define_hvx_intrinsic";
                     }
+                    internal_assert(fn)
+                        << "Unable to find scalar broadcast function. Did you forget to specify either hvx_64 or hvx_128?";
```

Comment:
This is certainly helpful. Should we instead put a more heavyweight check in the constructor? (https://github.com/halide/Halide/blob/master/src/CodeGen_Hexagon.cpp#L36)
```
user_assert(target.features_any_of({{Halide::Target::HVX_128, Halide::Target::HVX_64}), "Need to specify target feature hvx_64 or hvx_128");
```
after https://github.com/halide/Halide/blob/master/src/CodeGen_Hexagon.cpp#L52

Reply:
If it's generally true that one or the other is required for any codegen on HVX, then yes, absolutely.

Reply:
If we assert more generally, I think we should remove the "Did you forget..." here, because that will be unlikely to help.

Reply:
PTAL